### ISSUE TITLE
Sipe gesture to bring up Hacker's Keyboard

### ISSUE THREAD

**Issue by ronnac** [klausw/hackerskeyboard#636]
I have been using the permanent notification in order to bring up the Hacker's Keyboard whenever I need it outside a text editing field.

However, I didn't like the notification permanently taking up the space in the notification area.

What I prefer is to have a gesture (in my case, swipe up from the bottom with two fingers) that brings up Hacker's Keyboard.

Here is what I have done to make it work:

1) I bought "GMD Gesture Control" from PlayStore.
It allows to define a gesture in order to launch an application.

2) In "LatinIME.java" of Hacker's Keyboard I have added a static method:

```
public static void forceShow(){
	InputMethodManager imm = (InputMethodManager)
		sInstance.getApplicationContext().getSystemService(COntext.INPUT_METHOD_SERVICE);
	if (imm!=null) {
		imm.showSoftInputFromInputMethod(sInstance.mToken, InputMethodManager.SHOW_FORCED);
	}

}
```

3) I have added an Activity Class "ForceShow.java" to the package of Hacker's Keyboard. All it does is to launch the static method forceShow defined in step 2):

```
package org.pocketworkstation.pckeyboard;

import android.app.*;
import android.os.*;

public class ForceShow extends Activity {
    static final String TAG = "PCKeyboard/Notification";
	
    @Override
    protected void onCreate(Bundle savedInstanceState) {
       super.onCreate(savedInstanceState);
   	
       LatinIME.forceShow();
       this.finish();
    }
}
```

4) I have created an independent application "showKeyboard", that starts above Activity "ForceShow".

```
package com.niveauverleih.showkeyboard;

import android.app.*;
import android.content.*;
import android.os.*;
import android.view.*;

public class MainActivity extends Activity 
{
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
       
		Intent intentAux = new Intent("org.pocketworkstation.pckeyboard.ForceShow");
		startActivity(intentAux);

		finish();
    }
}

```
With the following AndroidManifest.xml, which makes the app "invisible" (the "NoDisplay" part of it):

```
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.niveauverleih.zzshowkeyboard" >

    <application
        android:allowBackup="true"
        android:icon="@drawable/ic_launcher"
        android:label="@string/app_name"
        android:theme="@style/AppTheme"
		android:resizeableActivity = "true">
        <activity
            android:name=".MainActivity"
            android:theme="@android:style/Theme.NoDisplay" >
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />g
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
    </application>
	
    <uses-permission android:name="android.permission.EXPAND_STATUS_BAR"/>
    <uses-permission android:name="android.permission.STATUS_BAR"/>

</manifest>

```

5) In "GMD Gesture Control" I have assigned the gesture "swipe two fingers up from bottom" to the application "showKeyboard".

To summarize, I can now force Hacker's Keyboard to show up by swiping 2 fingers up in any context.
This gesture triggers the launch of the app ""showKeyboard" (via "GMD Gesture Control") , which calls the new Activity "ForceShow", which in turn calls the static method forceShow.

Could you please verify it and give your comments? Would it be interesting to add this possibility to Hacker's Keyboard?

**Comment by rkd2003** [comment]
Oh...! Can you upload that app? I can't do it. I'm not developer...
